WireUI/Sources/WireIndividualToTeamMigrationUI/IndividualToTeamMigrationViewController.swift

```diff
@@ -0,0 +1,226 @@
+        case .dismissCancellationAlert:
+            childController.dismiss(animated: true)
+        case .toPlans:
+            let vc = hostedView(
+                for: .teamPlanSelection(features: features),
+                stepIndex: (childController.viewControllers.count + 1),
+                stepCount: 4,
+                target: self,
+                onTransition: { @MainActor [weak self] in self?.transition(to: $0) }
+            )
+            childController.pushViewController(vc, animated: false)
+        case .toTeamName:
+            let vc = hostedView(
+                for: .teamName,
+                stepIndex: (childController.viewControllers.count + 1),
+                stepCount: 4,
+                target: self,
+                onTransition: { @MainActor [weak self] in self?.transition(to: $0) }
+            )
+            childController.pushViewController(vc, animated: true)
+        case .toConfirmation:
+            let vc = hostedView(
+                for: .confirmation,
+                stepIndex: (childController.viewControllers.count + 1),
+                stepCount: 4,
+                target: self,
+                onTransition: { @MainActor [weak self] in self?.transition(to: $0) }
+            )
+            childController.pushViewController(vc, animated: true)
+        case .toCompletion:
+            let vc = hostedView(
+                for: .completion(profileName: "Profile Name", teamName: "Some Team"),
+                stepIndex: (childController.viewControllers.count + 1),
+                stepCount: 4,
+                target: self,
+                onTransition: { @MainActor [weak self] in self?.transition(to: $0) }
+            )
+            childController.pushViewController(vc, animated: true)
+        case .toApp:
+            break
+        case .toTeamManagement:
+            break
+        }
+    }
+
+    @objc func onDismissTapped() {
+        transition(to: .toCancellationAlert)
+    }
+}
+
+@MainActor
+private func hostedView(
+    for step: IndividualToTeamMigrationViewController.Step,
+    stepIndex: Int,
+    stepCount: Int,
+    target: IndividualToTeamMigrationViewController,
+    onTransition transitionCallback: @escaping @MainActor @Sendable (IndividualToTeamMigrationViewController.Transition) -> Void
+) -> UIViewController {
+    let vc = UIHostingController(rootView:
+        PageContainer(
+            content: { viewFor(
+                step: step,
+                stepIndex: stepIndex,
+                stepCount: stepCount,
+                onTransition: transitionCallback
+            ) },
+            step: stepIndex,
+            stepCount: stepCount,
+            stepTitle: step.title
+        )
+    )
+    vc.title = step.title
+    vc.navigationItem.rightBarButtonItem = UIBarButtonItem(
+        image: UIImage(systemName: "xmark"),
+        style: .plain,
+        target: target,
+        action: #selector(IndividualToTeamMigrationViewController.onDismissTapped)
```

Comment:
```
public extension UIBarButtonItem {
    /// Creates a customized close button for use in navigation bars.
    ///
    /// This method creates a UIBarButtonItem configured as a close button with a custom image,
    /// tint color, and accessibility features.
    ///
    /// - Parameters:
    ///   - action: A UIAction to be performed when the button is tapped.
    ///   - accessibilityLabel: A string describing the button's purpose for accessibility.
    ///
    /// - Returns: A UIBarButtonItem configured as a close button.
    static func closeButton(action: UIAction, accessibilityLabel: String) -> UIBarButtonItem {
        let closeImage = UIImage(named: "Close")

        let closeItem = UIBarButtonItem(title: accessibilityLabel, primaryAction: action)

        closeItem.image = closeImage

        closeItem.style = .plain
        closeItem.tintColor = SemanticColors.Icon.foregroundDefaultBlack

        closeItem.accessibilityLabel = accessibilityLabel
        closeItem.accessibilityIdentifier = "close"

        return closeItem
    }
}
```

We have this helper method creating this X button. the only thing you have to provide is a UIAction and accessibilityLabel. By using a UIAction we can get rid of the selector pattern and get rid of the @objc annotation from the dismiss method

Reply:
I can even get rid of the dismiss method altogether. Thanks!